### PR TITLE
Box more specified values to bring SpecifiedValue/PropertyDeclaration sizes down to 24/32 bytes

### PR DESCRIPTION
**Pull request by bholley**
I think these properties are generally rare enough that they're not worth adding 8 bytes of memmove overhead on every PropertyDeclaration. There are a _lot_ of 24-byte properties though, so I think that's probably a good sustainable level.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16015)

### CODE REVIEW

File: components/style/properties/properties.mako.rs [hunk 1]
```diff
@@ -2490,7 +2490,10 @@ pub fn test_size_of_specified_values() {
                         increasing the size may negative affect style system performance. Please consider \
                         using `boxed=\"True\"` in this longhand.",
                         specified_value.0, specified_value.1, threshold));
-        } else if specified_value.1 <= threshold && specified_value.2 {
+        } else if specified_value.1 <= threshold && specified_value.2 &&
+                  specified_value.0 != "cursor" /* Cursor needs to be boxed for Gecko but not Servo,
+                                                   and the declarative setup makes it hard to make
```
Comment (Manishearth) [inline components/style/properties/properties.mako.rs]: `boxed = "${product == 'gecko'}"` should work

**Review by Manishearth — APPROVED** [review]
r+, but try to make the cursor thing work

**Comment by bholley** [comment]
Thanks.

@bors-servo r=Manishearth

**Comment by Wafflespeanut** [comment]
Looks like `dom/element.rs` needs to be updated.

    error[E0308]: mismatched types
       --> /Users/servo/buildbot/slave/mac-dev-unit/build/components/script/dom/element.rs:626:53
        |
    626 |                 PropertyDeclaration::BorderTopWidth(width_value.clone())));
        |                                                     ^^^^^^^^^^^^^^^^^^^ expected struct `std::boxed::Box`, found enum `style::values::specified::BorderWidth`
        |
        = note: expected type `std::boxed::Box<style::values::specified::BorderWidth>`
                   found type `style::values::specified::BorderWidth`

    error[E0308]: mismatched types
       --> /Users/servo/buildbot/slave/mac-dev-unit/build/components/script/dom/element.rs:628:54
        |
    628 |                 PropertyDeclaration::BorderLeftWidth(width_value.clone())));
        |                                                      ^^^^^^^^^^^^^^^^^^^ expected struct `std::boxed::Box`, found enum `style::values::specified::BorderWidth`
        |
        = note: expected type `std::boxed::Box<style::values::specified::BorderWidth>`
                   found type `style::values::specified::BorderWidth`

    error[E0308]: mismatched types
       --> /Users/servo/buildbot/slave/mac-dev-unit/build/components/script/dom/element.rs:630:56
        |
    630 |                 PropertyDeclaration::BorderBottomWidth(width_value.clone())));
        |                                                        ^^^^^^^^^^^^^^^^^^^ expected struct `std::boxed::Box`, found enum `style::values::specified::BorderWidth`
        |
        = note: expected type `std::boxed::Box<style::values::specified::BorderWidth>`
                   found type `style::values::specified::BorderWidth`

    error[E0308]: mismatched types
       --> /Users/servo/buildbot/slave/mac-dev-unit/build/components/script/dom/element.rs:632:55
        |
    632 |                 PropertyDeclaration::BorderRightWidth(width_value)));
        |                                                       ^^^^^^^^^^^ expected struct `std::boxed::Box`, found enum `style::values::specified::BorderWidth`
        |
        = note: expected type `std::boxed::Box<style::values::specified::BorderWidth>`
                   found type `style::values::specified::BorderWidth`

    error: aborting due to 4 previous errors

    error: Could not compile `script`.

**Comment by emilio** [comment]
This is unfortunate. I think instead of boxing lengths we should be making `Length` smaller.

**Comment by emilio** [comment]
> This is unfortunate. I think instead of boxing lengths we should be making Length smaller.

This might not be as trivial as I initially thought though, since CalcLengthOrPercentage is already boxed.

**Comment by Manishearth** [comment]
I think that's part of the plan, too. It's possible this might be achieved by flattening out the length enum, perhaps into `LengthOr<A>` or something.

**Comment by bholley** [comment]
Definitely happy to unbox what we can if people have ideas. I've just been seeing a lot of memmov traffic and I think locking SpecifiedValue down to 24 bytes is a win.

**Pull request merged**